sdk/iot/core/tests/cmocka/az_iot_hub_client_tests.c

```diff
@@ -21,3 +36,175 @@ void test_az_iot_hub_client_get_default_options_succeed(void** state)
   assert_true(az_span_is_content_equal(options.module_id, AZ_SPAN_NULL));
   assert_true(az_span_is_content_equal(options.user_agent, AZ_SPAN_NULL));
 }
+
+void test_az_iot_hub_client_init_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_true(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  assert_string_equal(TEST_DEVICE_ID_STR, az_span_ptr(client._internal.device_id));
+  assert_string_equal(TEST_DEVICE_HOSTNAME_STR, az_span_ptr(client._internal.iot_hub_hostname));
+}
+
+void test_az_iot_hub_client_init_custom_options_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options;
+  options.module_id = AZ_SPAN_FROM_STR(TEST_MODULE_ID);
+  options.user_agent = AZ_SPAN_FROM_STR(TEST_USER_AGENT);
+  assert_true(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options) == AZ_OK);
+
+  assert_string_equal(TEST_DEVICE_ID_STR, az_span_ptr(client._internal.device_id));
+  assert_string_equal(TEST_DEVICE_HOSTNAME_STR, az_span_ptr(client._internal.iot_hub_hostname));
+  assert_string_equal(TEST_MODULE_ID, az_span_ptr(client._internal.options.module_id));
+  assert_string_equal(TEST_USER_AGENT, az_span_ptr(client._internal.options.user_agent));
+}
+
+void test_az_iot_hub_client_user_name_get_succeed(void** state)
```

Comment:
>test_az_iot_hub_client_user_name_get_succeed [](start = 5, length = 44)

Can we have bound check unit tests also ?

when we give memory with pattern in it and expect pattern is not erased outsize of capacity given to function?